### PR TITLE
Package source code as sumologic_collectd_metrics library

### PR DESCRIPTION
**Pull request by CCheSumo**
This PR packages the entire source code into a standard Python library and also specifies dependencies. The advantages of this change are

- No additional requirements of user have `git` ready in their production system
- No requirements for them to install dependent libraries. Pip install will take care of it.
- No requirements of setting up dir for storing the module source code and then specify the module path in config file
- Better exposure through python index

Lincoln brought up the point that user may not want to install another package apart from collectd. Instead of enabling pip install, just put a install script. Though it seems to me just an informal way of mimicking what pip install does with a lot more extra pain points. I do want to get the agreement to proceed with the PR before adding more efforts. 

What is left, 

- README mark up is not compatible with python release markup language, need to update it for the documentation to look nicely.
- Update README content to include the new setup
- Setup release repo and release with comments in this file https://docs.google.com/document/d/1kxF_8AKC6YStCQiBQVQf48tTg8He7zgXGWQz9D-tnDQ/edit addressed.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[metadata]
+description-file = README.md
```
Comment (CCheSumo) [inline setup.cfg]: README.md needs to be updated README.rst with correct markup

**Comment by CCheSumo** [comment]
Oh, I just realized that if a customer is reluctant to install it as a library. They always have the option to do git clone so there is no bad side of it.

**Comment by latkin** [comment]
I'm fine with exposing as a pip package, and I imagine some folks will be happy to consume it that way.

We can offer in more than one place, and observe to see how many downloads/engagements we get from each place.

IMO we can start with 2:

  - Available via pip (this PR)
  - Create a github Release that allows simple download of everything as gzip

**Comment by CCheSumo** [comment]
@rvmiller89 Yeah working on another PR to update the README now

**Pull request merged**